Document minimum supported platform/OS versions (#458)
powersync-ja/powersync-docs: resources/supported-platforms.mdx

| --- | --- |  --- |
| <Icon icon="elephant" iconType="solid" /> PostgreSQL | 11+ | |
| <Icon icon="leaf" iconType="solid" /> MongoDB | 6.0+ | |
| <Icon icon="dolphin" iconType="solid" /> MySQL | 5.7+ | |
| <Icon icon="server" iconType="solid" /> SQL Server (beta) | SQL Server 2019+ (15.0+) or Azure SQL Database | Beta release.<br />Azure SQL Database is always running on the latest version of the SQL Server DB Engine. |
| <Icon icon="database" iconType="solid" /> Convex (alpha) | — | Alpha release. |

## Supported SDK Platforms

Jump to: [<Icon icon="flutter" iconType="solid" /> Dart/Flutter SDK](#dart/flutter-sdk), [<Icon icon="react" iconType="solid" /> React Native SDK](#react-native-sdk), [<Icon icon="js" iconType="solid" /> Javascript SDK](#js/web-sdk), [<Icon icon="c" iconType="solid" /> Capacitor SDK](#capacitor-sdk), [<Icon icon="rust"  /> Tauri SDK](#tauri-sdk), [<Icon icon="node-js" /> Node.js SDK](#node-js-sdk), [<Icon icon="android" /> Kotlin SDK](#kotlin-sdk), [<Icon icon="swift" iconType="solid" /> Swift SDK](#swift-sdk), [<Icon icon="microsoft" iconType="solid" /> .NET SDK](#net-sdk), [<Icon icon="rust" /> Rust SDK](#rust-sdk)

### Widespread SDK Platform Support

> The versions below are the minimum supported versions across all PowerSync Client SDKs, so you can expect them to work regardless of which SDK you use.

Reviewer: Maybe point out that some client SDKs support older versions too. At least for me, this reads as if we don't support anything older than say macOS 14 in any SDK which is not true (Swift supports macOS 12, Flutter supports 10.15).

| Platform | Version |
| --- | --- |
| <Icon icon="android" /> Android | SDK 24 (Android 7.0) |
| <Icon icon="apple" /> iOS | 15.0 |
| <Icon icon="apple" /> macOS | 14.0 (Sonoma) |
| <Icon icon="windows" /> Windows | 10 |
| <Icon icon="linux" /> Linux | glibc 2.34+ |
| <Icon icon="js" /> Web | See [Web Compatibility](#web-compatibility). |

> ### Web Compatibility

Reviewer: I wonder if we should move this to a JavaScript subsection since the VFS names are not applicable to Dart?

Author: Moved `Web Compatibility` to a subheading of JS and removed the reference to it in `Dart/Flutter`.

| OPFSWriteAheadVFS   | ✅ | ✅ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |

### <Icon icon="flutter" iconType="solid" size={24}/> Dart/Flutter SDK

| Platform / Feature | Supported? | Platform Version | Notes |
| --- | --- | --- | --- |
| Android | Yes (x86-64, aarch64, armv7) | SDK 24 (Android 7.0) | Requires Flutter |
| iOS | Yes | 14.0 (UNTESTED) | Requires Flutter |
| macOS | Yes (x86-64, aarch64) | 14.0 (UNTESTED) | |
| Windows | Yes (x86-64, x86, aarch64) | 10 (UNTESTED) | |
| Linux | Yes (x86-64, x86, aarch64, armv7, riscv64gc) | Requires glibc 2.34+ | |
| Web | Yes | See [Web Compatibility](#web-compatibility). | Only dart2js is tested, dart2wasm has issues |
| HTTP connection method | Yes | | |
| WebSocket connection method | No | | |

### <Icon icon="react" iconType="solid" size={24}/> React Native SDK

| Platform / Feature | Supported? | Platform Version | Notes |
| --- | --- | --- | --- |
| React Native | Yes | | |
| React Native w/ Expo | Yes | | |
| React Native for Web | Yes | See [Web Compatibility](#web-compatibility). | |
| React Strict DOM | YMMV - not tested | | |
| React Native for Windows | No | | |
| HTTP connection method | Yes | | Legacy (supported) |

> | WebSocket connection method | Yes | | Default |

Reviewer: We can change it here or in the v2 docs branch, just wanted to flag that HTTP is no longer considered legacy and in fact the default. WebSocket is the workaround only used for React Native without Expo since we have no good HTTP client.

Author: I think we can change it again in v2, since I believe WebSockets is still the default currently (correct me if I'm wrong). I'll remove the Legacy tag from HTTP.

| iOS | Yes (aarch64 device and simulator) | | |
| macOS (native) | Yes (aarch64) | | |
| macOS catalyst (native) | No | | Blocked on [KT-40442: Support building Kotlin/Native for Mac Catalyst (x86-64 and arm64)](https://youtrack.jetbrains.com/issue/KT-40442/Support-building-Kotlin-Native-for-Mac-Catalyst-x86-64-and-arm64) |
| watchOS | Yes (aarch64 device, armv8 32-bit pointers ABI, aarch64 simulator) | | |
| tvOS | Yes (aarch64 device, aarch64 simulator) | | |
| visionOS | No | | Blocked on [KT-59571: Add support for visionOS SDK](https://youtrack.jetbrains.com/issue/KT-59571/Add-support-for-visionOS-SDK) |
| Windows (JVM) | Yes (x86-64 only) | | |
| Linux (JVM) | Yes (x86-64, aarch64) | | |
| macOS (JVM) | Yes (x86-64, aarch64) | | |
| Linux (native) | No | | Maybe soon |
| Windows (native) | No | | Maybe soon |
| JS | No | | |
| WebAssembly | No | | |
| HTTP connection method | Yes | | |
| WebSocket connection method | Yes | | Note: Only as an automated fallback for clients without backpressure support. |

### <Icon icon="swift" iconType="solid" size={24}/> Swift SDK

| Platform / Feature | Supported? | Platform Version | Notes |
| --- | --- | --- | --- |

> | macOS | Yes | | |

Reviewer: Perhaps add [these minimum versions](https://github.com/powersync-ja/powersync-swift/blob/b855d4997991186c60313d35919c4c813d7ccb3e/Package.swift#L36-L39)?

| iOS | Yes | | |

> | watchOS | Yes | | watchOS 26 not supported yet |

Reviewer: I believe this has been fixed and we support watchOS 26 again.

description: "Supported platforms and feature availability for each PowerSync Client SDK across mobile, web, and desktop."
---

## Widespread SDK Platform Support

The versions below are the minimum supported versions across all PowerSync Client SDKs, so you can expect them to work regardless of which SDK you use.

| Platform | Version |
| --- | --- |
| <Icon icon="android" /> Android | SDK 24 (Android 7.0) |
| <Icon icon="apple" /> iOS | 15.0 |
| <Icon icon="apple" /> macOS | 14.0 (Sonoma) |
| <Icon icon="windows" /> Windows | 10 |
| <Icon icon="linux" /> Linux | glibc 2.34+ |
| <Icon icon="js" /> Web | See [Web Compatibility](#web-compatibility). |

Note that some client SDKs may support lower versions than the ones listed here.

## Supported SDK Platforms

> > Note: An empty `Platform Version` cell means that the lowest target version is unknown/untested.

Reviewer 2: ```suggestion
> Note: An empty `Platform Version` cell means that the lowest target version is currently unknown/untested.
```

> > Note: An empty `Platform Version` column means that the lowest target version is unknown/untested.

Reviewer 2: ```suggestion
> Note: An empty `Platform Version` cell means that the lowest target version is unknown/untested.
```

description: "Supported platforms and feature availability for each PowerSync Client SDK across mobile, web, and desktop."
---

> Jump to: [Source Databases](#supported-source-databases), [SDK Platforms](#supported-sdk-platforms), [Supported Platforms Policy](#supported-platforms-policy), [SDK Versioning Policy](#sdk-versioning-policy)

Reviewer 2: We generally rely on the "On this page" breakdown on the side for this, this feels redundant (and inconsistent). If we want to give the user an idea of everything that this page covers (which is a lot), I think a introduction in words/bullets is better.

Reviewer 3: agreed we probably don't need this

## Supported Source Databases 

Each supported source database has a minimum required version to be compatible with PowerSync.

| Database | Version |  Notes |
| --- | --- |  --- |
| <Icon icon="elephant" iconType="solid" /> PostgreSQL | 11+ | |
| <Icon icon="leaf" iconType="solid" /> MongoDB | 6.0+ | |
| <Icon icon="dolphin" iconType="solid" /> MySQL | 5.7+ | |
| <Icon icon="server" iconType="solid" /> SQL Server (beta) | SQL Server 2019+ (15.0+) or Azure SQL Database | Beta release.<br />Azure SQL Database is always running on the latest version of the SQL Server DB Engine. |
| <Icon icon="database" iconType="solid" /> Convex (alpha) | — | Alpha release. |

## Supported SDK Platforms

> Jump to: [<Icon icon="flutter" iconType="solid" /> Dart/Flutter SDK](#dart/flutter-sdk), [<Icon icon="react" iconType="solid" /> React Native SDK](#react-native-sdk), [<Icon icon="js" iconType="solid" /> Javascript SDK](#js/web-sdk), [<Icon icon="c" iconType="solid" /> Capacitor SDK](#capacitor-sdk), [<Icon icon="rust"  /> Tauri SDK](#tauri-sdk), [<Icon icon="node-js" /> Node.js SDK](#node-js-sdk), [<Icon icon="android" /> Kotlin SDK](#kotlin-sdk), [<Icon icon="swift" iconType="solid" /> Swift SDK](#swift-sdk), [<Icon icon="microsoft" iconType="solid" /> .NET SDK](#net-sdk), [<Icon icon="rust" /> Rust SDK](#rust-sdk)

Reviewer 2: Similar comment as above. I get why we have it - there's a lot of content here, but if you're worried about excessive scrolling maybe rather consider accordions (a collapsible section) for each SDK like we have in https://docs.powersync.com/intro/examples?

Reviewer 3: probably don't need either a jump to or an accordion. it's ok to scroll this page IMO since it's typically not on the golden path

### Widespread SDK Platform Support

The versions below are the minimum supported versions across all PowerSync Client SDKs, so you can expect them to work regardless of which SDK you use.

Note: Some client SDKs may support lower versions than the ones listed here.

| Platform | Version |
| --- | --- |
| <Icon icon="android" /> Android | SDK 24 (Android 7.0) |
| <Icon icon="apple" /> iOS | 15.0 |
| <Icon icon="apple" /> macOS | 14.0 (Sonoma) |
| <Icon icon="windows" /> Windows | 10 |
| <Icon icon="linux" /> Linux | glibc 2.34+ |
| <Icon icon="js" /> Web | See [Web Compatibility](#web-compatibility). |

> ### <Icon icon="flutter" iconType="solid" size={24}/> Dart/Flutter SDK

Reviewer 2: I think just above these SDK sections we should add a callout or something stating that if a Platform Version isn't indicated (e.g. there are a lot of empty cells in the "Platform Version" column in these tables below) it means it's as of yet unspecified or untested or unknown - not sure what the correct term here is and users should refer to Widespread SDK table for now. i.e. that these tables below are a WIP?

Reviewer 2: On the same note, if an entire table does not ahve a value in that column (e.g. Tauri) I think let's leave it out for now - it looks a bit jarring being just and empty column, or add some placeholder content there.

### <Icon icon="microsoft" iconType="solid" size={24}/> .NET SDK

| Platform / Feature | Supported? | Platform Version | Notes |
| --- | --- | --- | --- |
| WPF | Yes | Windows 10 | |
| MAUI for Android | Yes | SDK 24 (Android 7.0) | |
| MAUI for iOS | Yes | 14.0 | |
| MAUI for Windows | Yes | 10 | |
| MAUI for Mac Catalyst | Yes | Catalyst 17.0 (macOS 14.0) | Added in `PowerSync.Maui` 0.1.2. |
| Winforms | YMMV - not tested | | |
| CLI Windows | Yes (x64, ARM) | | |
| CLI Mac | Yes (x64, ARM) | | |
| CLI Linux | Yes (x64, ARM) | | |
| Avalonia UI | YMMV - not tested | | |
| HTTP connection method | Yes | | |
| WebSocket connection method | No | | |

### <Icon icon="rust" iconType="solid" size={24}/> Rust SDK

| Platform / Feature | Supported? | Notes |
| --- | --- | --- |
| All | Yes | The SDK supports all `std` Rust targets, but is currently only tested on Linux. |
| HTTP connection method | Yes | |
| WebSocket connection method | No | |

## Supported Platforms Policy

PowerSync supports a wide number of target platforms across its many SDKs. We aim to support as many platforms and versions as are useful for users and practical to maintain.

> ### When we raise the minimum version

Reviewer 4: Should we mention that we generally raise the minimum version of an OS only in new major SDK versions? Or is that not really feasible in all cases?

This would specifically apply for things like Android/iOS/Windows or DB versions, not something that the user can easily upgrade like NodeJS/Flutter/RN versions.

cc @simolus3 / @Chriztiaan - you may have more experience in what is practical for us here.

Reviewer: I'm leaning towards this not being feasible in all cases. Using AndroidX as an example, they don't use major versions to raise their minimum Android SDK versions. So at that point we'd have to either stop updating our dependencies or cut a major release ourselves. Another example is Kotlin/Native moving towards deprecating X64 Apple targets, we had to drop them too to support the latest dependencies.

Reviewer 4: Ok, happy to just leave that out in that case.

In general, PowerSync aims to support any platform version that still receives security updates from the platform provider. For platforms that follow an LTS and Current release model, this means we aim to support all active LTS releases along with the Current release.

> Supporting every version that fits the above criteria is [not always feasible](#exceptions). When we cannot support all of them, we prioritise support for newer versions over older versions to ease maintenance. We will occasionally maintain support for an older version if an enterprise customer requires it. 

Reviewer 3: ```suggestion
Supporting every version that fits the above criteria is [not always feasible](#exceptions). When we cannot support all of them, we prioritise support for newer versions over older versions to ease maintenance. We will occasionally maintain support for an older version on a case-by-case basis for larger customers. Reach out to us at `hello@powersync.com` if this is you. 
```
